atom/browser/browser.cc

```diff
@@ -154,10 +154,23 @@ void Browser::DidFinishLaunching(const base::DictionaryValue& launch_info) {
     base::CreateDirectoryAndGetError(user_data, nullptr);
 
   is_ready_ = true;
+  if (ready_promise_) {
+    ready_promise_->Resolve();
+  }
   for (BrowserObserver& observer : observers_)
     observer.OnFinishLaunching(launch_info);
 }
 
+util::Promise* Browser::WhenReady(v8::Isolate* isolate) {
+  if (!ready_promise_) {
+    ready_promise_ = new util::Promise(isolate);
```

Comment:
It looks like this leaks -- is it freed somewhere that I'm not seeing?

Reply:
The existing JS impl leaked it as well, it's effectively a permanently memoized getter